File: dotnet_3/cs/rest/SCS/Imp/Title.cs
namespace SCS.Imp
{
    public class Title
    {
        public static string Subject(string sex, string title)
        {
            //CHECK PERSONAL TITLE CONSISTENT WITH SEX
            sex = sex.ToLower();
            title = title.ToLower();
            var result = -1;

            if (sex == "male") {
                if ("mr".Equals(title) ||
                    "dr".Equals(title) ||
                    "sir".Equals(title) ||
                    "rev".Equals(title) ||
                    "rthon".Equals(title) ||
                    "prof".Equals(title)) {
                    result = 1;
                }
            }
            else if (sex == "female") {
                if ("mrs".Equals(title) ||
                    "miss".Equals(title) ||
                    "ms".Equals(title) ||
                    "dr".Equals(title) ||
                    "lady".Equals(title) ||
                    "rev".Equals(title) ||
                    "rthon".Equals(title) ||
                    "prof".Equals(title)) {
                    result = 0;
                }
            }
            else if (sex == "none") {
                if ("dr".Equals(title) ||
                    "rev".Equals(title) ||
                    "rthon".Equals(title) ||
                    "prof".Equals(title)) {
                    result = 2;
                }
            }

            return "" + result;
        }
    }
}

File: jdk_8_maven/cs/graphql/graphql-scs/src/main/kotlin/org/graphqlscs/type/Title.kt
package org.graphqlscs.type

import org.springframework.stereotype.Component
import java.util.*
@Component
class Title {

    fun subject(sex: String, title: String): String {
        //CHECK PERSONAL TITLE CONSISTENT WITH SEX
        var sex = sex
        var title = title
        sex = sex.lowercase(Locale.getDefault())
        title = title.lowercase(Locale.getDefault())
        var result = -1
        if ("male" == sex) {
            if ("mr" == title || "dr" == title || "sir" == title || "rev" == title || "rthon" == title || "prof" == title) {
                result = 1
            }
        } else if ("female" == sex) {
            if ("mrs" == title || "miss" == title || "ms" == title || "dr" == title || "lady" == title || "rev" == title || "rthon" == title || "prof" == title) {
                result = 0
            }
        } else if ("none" == sex) {
            if ("dr" == title || "rev" == title || "rthon" == title || "prof" == title) {
                result = 2
            }
        }
        return "" + result
    }
}